airbyte-server/src/main/java/io/airbyte/server/handlers/ArchiveHandler.java

```diff
@@ -0,0 +1,181 @@
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ArchiveHandler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ArchiveHandler.class);
+  private static final String ARCHIVE_FILE_NAME = "airbyte_config_data";
+  private static final String VERSION_FILE_NAME = "VERSION";
+
+  private final String version;
+  private final ConfigRepository configRepository;
+
+  public ArchiveHandler(final String version, final ConfigRepository configRepository) {
+    this.version = version;
+    this.configRepository = configRepository;
+  }
+
+  public Path exportData() {
+    try {
+      final Path tempFolder = Files.createTempDirectory("airbyte_archive");
+      final Path archive = Files.createTempFile(ARCHIVE_FILE_NAME, ".tar.gz");;
+      archive.toFile().deleteOnExit();
+      try {
+        exportVersionFile(tempFolder);
+        exportAirbyteConfig(tempFolder);
+        exportAirbyteDatabase(tempFolder);
+        Archives.createArchive(tempFolder, archive);
+      } finally {
+        FileUtils.deleteDirectory(tempFolder.toFile());
+      }
+      return archive;
+    } catch (IOException e) {
+      LOGGER.error("Export Data failed.");
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void exportVersionFile(Path tempFolder) throws IOException {
+    final File versionFile = Files.createFile(tempFolder.resolve(VERSION_FILE_NAME)).toFile();
+    FileUtils.writeStringToFile(versionFile, version, Charset.defaultCharset());
+  }
+
+  private void exportAirbyteConfig(Path tempFolder) {
+    LOGGER.info("Exporting Airbyte Configs");
+    final ConfigFileArchiver configFileArchiver = new ConfigFileArchiver(version, tempFolder);
+    Exceptions.toRuntime(() -> {
+      configFileArchiver.writeConfigsToArchive(ConfigSchema.STANDARD_WORKSPACE,
+          List.of(configRepository.getStandardWorkspace(PersistenceConstants.DEFAULT_WORKSPACE_ID)));
+      configFileArchiver.writeConfigsToArchive(ConfigSchema.STANDARD_SOURCE_DEFINITION, configRepository.listStandardSources());
+      configFileArchiver.writeConfigsToArchive(ConfigSchema.STANDARD_DESTINATION_DEFINITION, configRepository.listStandardDestinationDefinitions());
+      configFileArchiver.writeConfigsToArchive(ConfigSchema.SOURCE_CONNECTION, configRepository.listSourceConnection());
+      configFileArchiver.writeConfigsToArchive(ConfigSchema.DESTINATION_CONNECTION, configRepository.listDestinationConnection());
+      final List<StandardSync> standardSyncs = configRepository.listStandardSyncs();
+      configFileArchiver.writeConfigsToArchive(ConfigSchema.STANDARD_SYNC, standardSyncs);
+      final List<StandardSyncSchedule> standardSchedules = standardSyncs
```

Comment:
This is a really good demonstration of why schedules should stored as nested object of standard sync instead of as a separate entity.